udata/harvest/backends/base.py

```diff
@@ -168,6 +168,16 @@ def harvest(self):
         self.job = factory(status="initialized", started=datetime.utcnow(), source=self.source)
 
         before_harvest_job.send(self)
+        # Set harvest_activity_user_id on global context during the run
+        if current_app.config["HARVEST_ACTIVITY_USER_ID"]:
+            try:
+                # Try to fetch the existing harvest activity user
+                User.objects.get(id=current_app.config["HARVEST_ACTIVITY_USER_ID"])
+                g.harvest_activity_user_id = current_app.config["HARVEST_ACTIVITY_USER_ID"]
```

Comment:
Why not storing the user inside `g` to avoid getting it again in activity code?

Reply:
True! I guess it's cause I've added the check of the user existence in `backends/base.py` in a second iteration only.

Stored the user directly in https://github.com/opendatateam/udata/pull/3412/commits/1b3c857c13b0b59ed93e96d848b0d17b14f61ebd.